wsagent/che-core-api-project/src/main/java/org/eclipse/che/api/project/server/type/InitBaseProjectTypeHandler.java

```diff
@@ -42,7 +43,10 @@ public void onProjectInitialized(ProjectRegistry projectRegistry, FolderEntry pr
                                                                                                         NotFoundException {
         List<String> projects = projectRegistry.getProjects(projectFolder.getPath().toString());
         for (String project : projects) {
-            projectRegistry.setProjectType(project, BaseProjectType.ID, false);
+            RegisteredProject detected = projectRegistry.getProject(project);
```

Comment:
Why variable name is `detected` if you don't know there whether this project is detected or not?